### PR TITLE
items: better call_number search

### PR DESCRIPTION
**Pull request by rerowep**
* Adds `call_numbers` index with all call numbers.
* Adds `issue.inherited_second_call_number`.
* Closes: #3620.

### CODE REVIEW

**Review by lauren-d — APPROVED** [review]

File: rero_ils/modules/items/mappings/v7/items/item-v0.0.1.json [hunk 1]
```diff
@@ -39,8 +39,17 @@
         "analyzer": "autocomplete",
         "search_analyzer": "standard"
       },
+      "call_numbers": {
+        "type": "text",
+        "fields": {
+          "raw": {
+            "type": "keyword"
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_ils/modules/items/mappings/v7/items/item-v0.0.1.json:46]: This is a list of call numbers. `call_numbers` for string search and `call_numbers.raw` for keyword search.

**Review by Garfield-fr — APPROVED** [review]

File: rero_ils/modules/items/api/issue.py [hunk 2]
```diff
@@ -116,6 +116,18 @@ def issue_inherited_first_call_number(self):
...
             return holding.get('call_number')
 
+    @property
+    def issue_inherited_second_call_number(self):
+        """Get issue inherited second call number.
+
+        When the issue second call number is missing,
+        it returns the parent holdings second call number if exists.
+        """
+        from rero_ils.modules.holdings.api import Holding
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_ils/modules/items/api/issue.py:126]: Not possible. Circular include!

**Review by PascalRepond — APPROVED** [review]

**Comment by PascalRepond** [comment]
⚠️ Will probably require an update mapping and reindexing all items.

**Comment by PascalRepond** [comment]
## Tests

- [ ] REP [not directly linked]: If I edit a holdings and add a call number, the linked items are not reindexed with the call number.
    - See https://ilsdev.test.rero.ch/professional/records/holdings/detail/2642, the items do not contain the inherited call_number

**Pull request merged**